app/controllers/welcome_controller.rb

```diff
@@ -1,10 +1,5 @@
 class WelcomeController < ApplicationController
   def index
-    if params[:tag]
-      tagged_tutorials = Tutorial.tagged_with(params[:tag])
-      @tutorials = tagged_tutorials.paginate(page: params[:page], per_page: 5)
-    else
-      @tutorials = Tutorial.all.paginate(page: params[:page], per_page: 5)
-    end
+    @tutorials = Tutorial.filtered_tutorials(params, current_user)
```

Comment:
moved all this to the model where I called a few new methods

Reply:
![](https://media2.giphy.com/media/llxMTUf5tXK4nSo8ch/giphy.gif?cid=5a38a5a205d29eb20240543c5fe015f0933185a05c8b8c5d&rid=giphy.gif)